### PR TITLE
Update strawberry

### PR DESCRIPTION
**Pull request by marcoacierno**
Update strawberry to latest version

### CODE REVIEW

File: backend/api/blog/types.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import strawberry
 from api.users.types import User
-from strawberry.types.datetime import DateTime
+from strawberry.schema.types.base_scalars import DateTime
```
Comment (patrick91) [inline backend/api/blog/types.py:5]: I'd use datetime from python directly :)

File: backend/api/events/types.py [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
 
-if TYPE_CHECKING:  # pragma: no cover
-    from api.conferences.types import Conference
-
 
 @strawberry.type
 class Event:
     id: strawberry.ID
-    conference: "Conference"
+    conference: LazyType["Conference", "api.conferences.types"]
```
Comment (patrick91) [inline backend/api/events/types.py:15]: can you try with:

```suggestion
    conference: "api.conferences.types.Conference"
```

just curious to see if this works, maybe LazyType is actually not needed :)

File: backend/strawberry_forms/converter.py [hunk 3]
```diff
@@ -33,15 +33,15 @@ def type_or_optional_wrapped(type_, required):
 def convert_form_field_to_string(field):
     return (
         type_or_optional_wrapped(str, field.required),
-        strawberry.field(description=field.help_text, is_input=True),
+        strawberry.input(description=field.help_text),
```
Comment (patrick91) [inline backend/strawberry_forms/converter.py:36]: nnot sure this is right, why did you change it?

File: backend/strawberry_forms/mutations.py [hunk 4]
```diff
@@ -101,9 +102,11 @@ def mutate(root, info, input: input_type) -> output:
             def mutate(root, info) -> output:
                 return _mutate(root, info, {})
 
-        mutate.__name__ = f"{name}Output"
-        mutate.name = name
-        mutate = strawberry.mutation(mutate, description=cls.__doc__)
+        mutate = strawberry.mutation(
+            mutate,
+            name=f'{name[0].lower()}{name[1:]}',
+            description=cls.__doc__
```
Comment (patrick91) [inline backend/strawberry_forms/mutations.py:108]: let's not use descriptioon from the doc string ;)

File: backend/api/events/types.py (hunk 2 above)
Comment (marcoacierno) [reply backend/api/events/types.py:15]: it works but I need to `import api` at the top, I am not sure I like it 🤔

File: backend/strawberry_forms/converter.py [hunk 5]
```diff
@@ -33,15 +35,15 @@ def type_or_optional_wrapped(type_, required):
 def convert_form_field_to_string(field):
     return (
         type_or_optional_wrapped(str, field.required),
-        strawberry.field(description=field.help_text, is_input=True),
+        dataclasses.field(default=field.initial),
```
Comment (marcoacierno) [inline backend/strawberry_forms/converter.py:38]: we lose the `help_text` support here because we need to set the default value coming from the form, but as I plan on removing the forms integration anyway I think that's ok

File: backend/strawberry_forms/mutations.py [hunk 6]
```diff
@@ -101,9 +104,10 @@ def mutate(root, info, input: input_type) -> output:
             def mutate(root, info) -> output:
                 return _mutate(root, info, {})
 
-        mutate.__name__ = f"{name}Output"
-        mutate.name = name
-        mutate = strawberry.mutation(mutate, description=cls.__doc__)
+        # Hack because something changed and now the name contains `Mutation`
+        # as the classname, which makes sense but why it didn't happen before? lol
+        name = name.replace("Mutation", "")
```
Comment (marcoacierno) [inline backend/strawberry_forms/mutations.py:109]: so the name of the class is `ResetPasswordMutation`, assigned to `reset_password = ResetPasswordMutation`

Before the update of strawberry the mutation name used to be `resetPassword` but now it is `resetPasswordMutation`

This is a quick hack as I plan on removing the django forms integration anyway :)

File: backend/users/tests/test_reset_password_flow.py [hunk 7]
```diff
@@ -138,7 +138,7 @@ def test_cannot_reset_password_of_an_unkown_user(user_factory, graphql_client):
     user = user_factory(password="old")
     token = default_token_generator.make_token(user)
 
-    response = _reset_password(graphql_client, token, 5, "another")
+    response = _reset_password(graphql_client, token, 0, "another")
```
Comment (marcoacierno) [inline backend/users/tests/test_reset_password_flow.py:141]: this test seems wrong, to use `5` like 🤷 , these tests need a refactor as well

File: backend/strawberry_forms/tests/test_mutations.py [hunk 8]
```diff
@@ -1,73 +0,0 @@
-import strawberry
```
Comment (marcoacierno) [inline backend/strawberry_forms/tests/test_mutations.py:1]: I didn't manage to get the mutation to work, if someone has any idea happy to hear it, otherwise, I would just delete this code and move on removing the rest of strawberry_forms

File: backend/api/events/types.py (hunk 2 above)
Comment (patrick91) [reply backend/api/events/types.py:15]: you can hide it under `if TYPE_CHECKING`, anyway both solutions are ok for now :)

**Review by patrick91 — APPROVED** [review]
